File: app/src/main/java/com/themobilecompany/memorygame/ui/BaseActivity.java
package com.themobilecompany.memorygame.ui;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.themobilecompany.memorygame.MemoryGameApplication;
import com.themobilecompany.memorygame.injection.component.UiComponent;

public abstract class BaseActivity extends AppCompatActivity {

    private UiComponent uiComponent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        uiComponent = MemoryGameApplication.get(this).getComponent();
    }

    public UiComponent activityComponent() {
        return uiComponent;
    }

}

File: app/src/main/java/com/games/memorygame/ui/BaseActivity.kt
package com.games.memorygame.ui

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

import com.games.memorygame.MemoryGameApplication
import com.games.memorygame.injection.component.UiComponent

abstract class BaseActivity : AppCompatActivity() {

    private lateinit var uiComponent: UiComponent

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        uiComponent = MemoryGameApplication.get(this).component
    }

    fun activityComponent(): UiComponent = uiComponent

}
